src/main/java/uk/gov/hmcts/reform/sscs/service/NotificationService.java

```diff
@@ -50,54 +51,47 @@ public void manageNotificationAndSubscription(NotificationWrapper notificationWr
         log.info("Notification event triggered {} for case id {}", notificationType.getId(), caseId);
         for (SubscriptionWithType subscriptionWithType :
                 notificationWrapper.getSubscriptionsBasedOnNotificationType()) {
-            sendNotificationPerSubscription(notificationWrapper, subscriptionWithType, notificationType);
+            sendNotificationPerSubscription(notificationWrapper, subscriptionWithType.getSubscription(), notificationType, subscriptionWithType);
         }
     }
 
     private void sendNotificationPerSubscription(NotificationWrapper notificationWrapper,
-                                                 SubscriptionWithType subscriptionWithType,
-                                                 NotificationEventType notificationType) {
-        if (isValidNotification(notificationWrapper, subscriptionWithType.getSubscription(), notificationType)) {
+                                                 Subscription subscription,
+                                                 NotificationEventType notificationType,
+                                                 SubscriptionWithType subscriptionWithType) {
+        if (isValidNotification(notificationWrapper, subscriptionWithType, notificationType)) {
             Notification notification = notificationFactory.create(notificationWrapper,
                     subscriptionWithType.getSubscriptionType());
             if (notificationWrapper.getNotificationType().isAllowOutOfHours() || !outOfHoursCalculator.isItOutOfHours()) {
-                sendNotificationService.sendEmailSmsLetterNotification(notificationWrapper, subscriptionWithType.getSubscription(), notification);
-                processOldSubscriptionNotifications(notificationWrapper, notification);
+                sendNotificationService.sendEmailSmsLetterNotification(notificationWrapper, subscription, notification, subscriptionWithType);
+                processOldSubscriptionNotifications(notificationWrapper, notification, subscriptionWithType);
             } else {
                 notificationHandler.scheduleNotification(notificationWrapper);
             }
             reminderService.createReminders(notificationWrapper);
         }
     }
 
-    private boolean isValidNotification(NotificationWrapper wrapper, Subscription
-            subscription, NotificationEventType notificationType) {
-        return (isMandatoryLetter(notificationType) || (subscription != null && subscription.doesCaseHaveSubscriptions()
-                && notificationValidService.isNotificationStillValidToSend(wrapper.getNewSscsCaseData().getHearings(), notificationType)
-                && notificationValidService.isHearingTypeValidToSendNotification(wrapper.getNewSscsCaseData(), notificationType)));
+    private boolean isValidNotification(NotificationWrapper wrapper, SubscriptionWithType
```

Comment:
this if statement looks like is doing quite a few things. I'd make sure the different branches are well junit covered